pkg/admission/validate_bucketclass.go

```diff
@@ -0,0 +1,207 @@
+	if err := json.Unmarshal(rawBS, &BC); err != nil {
+		bcv.Logger.Error("error deserializing bucketclass")
+	}
+	return &BC
+}
+
+// ValidateCreate runs all the validations tests for CREATE operations
+func (bcv *BucketClassValidator) ValidateCreate() {
+	bcv.BucketClass = bcv.DeserializeBC(bcv.arRequest.Request.Object.Raw)
+
+	if ok, message := bcv.ValidateQuota(); !ok {
+		bcv.SetValidationResult(ok, message)
+		return
+	}
+	if bcv.BucketClass.Spec.NamespacePolicy != nil {
+		if ok, message := bcv.ValidateNSFSSingleBC(); !ok {
+			bcv.SetValidationResult(ok, message)
+			return
+		}
+	} else {
+		if ok, message := bcv.ValidateTiersNumber(); !ok {
+			bcv.SetValidationResult(ok, message)
+			return
+		}
+	}
+}
+
+// ValidateDelete runs all the validations tests for DELETE operations
+func (bcv *BucketClassValidator) ValidateDelete() {
+	bcv.BucketClass = bcv.DeserializeBC(bcv.arRequest.Request.OldObject.Raw)
+	if ok, message := bcv.ValidateBucketclassDeletion(); !ok {
+		bcv.SetValidationResult(ok, message)
+		return
+	}
+}
+
+// ValidateTiersNumber validates the provided number of tiers is 1 or 2
+func (bcv *BucketClassValidator) ValidateTiersNumber() (bool, string) {
+	numTiers := len(bcv.BucketClass.Spec.PlacementPolicy.Tiers)
+	if numTiers != 1 && numTiers != 2 {
+		return false, "Unsupported number of tiers, bucketClass supports only 1 or 2 tiers"
+	}
+	return true, "allowed"
+}
+
+// ValidateNSFSSingleBC validates that bucketclass configured to NS of type NSFS it will only be of type Single.
+func (bcv *BucketClassValidator) ValidateNSFSSingleBC() (bool, string) {
+	nspType := bcv.BucketClass.Spec.NamespacePolicy.Type
+	var namespaceStoresArr []string
+	if nspType == nbv1.NSBucketClassTypeMulti {
+		namespaceStoresArr = append(bcv.BucketClass.Spec.NamespacePolicy.Multi.ReadResources,
+			bcv.BucketClass.Spec.NamespacePolicy.Multi.WriteResource)
+	} else if nspType == nbv1.NSBucketClassTypeCache {
+		namespaceStoresArr = append(namespaceStoresArr, bcv.BucketClass.Spec.NamespacePolicy.Cache.HubResource)
+	} else {
+		return true, "allowed"
+	}
+	// check that namespace stores exists and their phase it ready
+	for _, name := range namespaceStoresArr {
+		nsStore := &nbv1.NamespaceStore{
+			TypeMeta: metav1.TypeMeta{Kind: "NamespaceStore"},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: bcv.BucketClass.Namespace,
+			},
+		}
+		if !util.KubeCheck(nsStore) {
+			bcv.Logger.Error("Failed to check namespacestore in NSFS single bucketclass type validation")
+			return true, "allowed"
+		}
+		if nsStore.Spec.Type == nbv1.NSStoreTypeNSFS {
+			return false, fmt.Sprintf("Invalid namespaceStore types, NSFS NamespaceStore %q is allowed on bucketclass of type Single", name)
+		}
+	}
+	return true, "allowed"
+}
+
+// ValidateQuota validates the quota configuration of the bucket class if provided
+func (bcv *BucketClassValidator) ValidateQuota() (bool, string) {
+	err := bucketclass.ValidateBucketClass(bcv.BucketClass)
```

Comment:
instead of calling validateBucketClass you should call validateQuotaConfig, in the future validateBucketClass will contain more validations..
which makes me think, why not add the validations to the bucketclass validation file and just call it ?